source_integration_test.go

```diff
@@ -153,6 +155,21 @@ func TestSource_EmptyTable(t *testing.T) {
 	_, err = source.Read(ctx)
 	is.True(errors.Is(err, sdk.ErrBackoffRetry))
 
+	// test CDC after an empty snapshot.
+	err = insertRecord(ctx, client, testTable, 0, 1)
+	is.NoErr(err)
+
+	for {
+		rec, err := source.Read(ctx)
+		if errors.Is(err, sdk.ErrBackoffRetry) {
+			continue
+		}
```

Comment:
If there's a bug in the source, we could get into an infinite loop here. So I believe it would be good to add a timeout to the ctx and run source.Read() in a goroutine.